### PR TITLE
Eth oracles gov

### PR DESCRIPTION
**Pull request by AnExsomnis**
WIP PR

### CODE REVIEW

File: core/types/data_source_external.go [hunk 1]
```diff
@@ -33,15 +37,34 @@ func (e *DataSourceDefinitionExternal) IntoProto() *vegapb.DataSourceDefinitionE
...
+			if dsn.EthOracle != nil {
+				return dsn.EthOracle.String()
+			}
+		}
 	}
 
 	return ""
 }
 
 func (e *DataSourceDefinitionExternal) DeepClone() dataSourceType {
```
Comment (ettec) [inline core/types/data_source_external.go:55]: BTW why do we do all this deep cloning, aren't these specs effectively immutable?

**Review by ettec — APPROVED** [review]

File: core/oracles/adaptors/ethereum.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"code.vegaprotocol.io/vega/core/oracles"
+)
+
+type EthereumAdaptor struct{}
+
+func NewEthereumAdaptor() *EthereumAdaptor {
+	return &EthereumAdaptor{}
+}
+
+f//unc (a *EthereumAdaptor) Normalise(data []byte, _ ...crypto.PublicKey) (*oracles.OracleData, error) {
```
Comment (pscott31) [inline core/oracles/adaptors/ethereum.go:15]: looks a bit borked :)

File: core/evtforward/ethcall/spec.go [hunk 3]
```diff
@@ -4,24 +4,26 @@ import (
...
+	types.Normaliser
 	requiredConfirmations uint64
 }
 
-func NewSpec(call Call, trigger Trigger, filter Filter, normaliser Normaliser, requiredConfirmations uint64) Spec {
+func NewSpec(call Call, trigger types.EthCallTrigger, filter types.Filter, normaliser types.Normaliser, requiredConfirmations uint64) Spec {
 	return Spec{
 		Call:                  call,
-		Trigger:               trigger,
+		EthCallTrigger:        trigger,
```
Comment (pscott31) [inline core/evtforward/ethcall/spec.go:26]: why the rename?

File: core/oracles/adaptors/ethereum.go (hunk 2 above)
Comment (AnExsomnis) [reply core/oracles/adaptors/ethereum.go]: Corrected thanks, not sure of that part is to stay anyway

File: core/types/data_source_external.go (hunk 1 above)
Comment (AnExsomnis) [reply core/types/data_source_external.go:55]: Removed them now, we do not need them actually, thanks!

File: protos/sources/vega/data_source.proto [hunk 4]
```diff
@@ -39,7 +39,7 @@ message DataSourceDefinitionExternal {
   // Types of External data sources
   oneof source_type {
     DataSourceSpecConfiguration oracle = 1;
-    EthCallSpec eth_call = 2;
+    EthCallSpec eth_oracle = 2;
```
Comment (candida-d) [inline protos/sources/vega/data_source.proto:42]: Docs please! :)

File: protos/sources/vega/data_source.proto [hunk 5]
```diff
@@ -101,9 +101,9 @@ message EthFilter {
   repeated vega.data.v1.Filter filters = 1;
 }
 
-message EthNormaliser {
+message Normaliser {
```
Comment (candida-d) [inline protos/sources/vega/data_source.proto:104]: Docs here too please

File: protos/sources/vega/data_source.proto (hunk 4 above)
Comment (ettec) [reply protos/sources/vega/data_source.proto:43]: this is merge to feature branch for us to coordinate work - there will be a PR to push this to develop in which we will be sure to sort the docs, its all a bit in flux at the moment as to how we exactly define these specs

**Pull request merged**